frontend/lib/data/user_repository.py

```diff
@@ -32,3 +32,40 @@ def list(self):
             response = self.table.scan(ExclusiveStartKey=response["LastEvaluatedKey"])
             data.extend(response["Items"])
         return data
+
+
+class RemoteUserRepository:
+    """
+    This will handle all the CRUD actions for users
+    in the AWS cognito user pool
+    """
+
+    def __init__(self, endpoint_url: str, region_name: str, identity_pool_id: str):
+        self.identity_pool_id = identity_pool_id
+        self.provider_name = (
+            f"cognito-idp.{region_name}.amazonaws.com/{identity_pool_id}"
+        )
+        self.cognito = boto3.client(
+            "cognito-identity", endpoint_url=endpoint_url, region_name=region_name
+        )
+
+    def get_or_create(self, username: str, password: str) -> dict:
```

Comment:
what is the scenario for get_or_create? I think we only need create, delete, get, and list operation

Reply:
The AWS SDK gives us a method that can get/retrieve an identity or create it if it's not there.  

From the [doc](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cognito-identity.html#CognitoIdentity.Client.get_id)
> Generates (or retrieves) a Cognito ID. Supplying multiple logins will create an implicit linked account. This is a public API. You do not need any credentials to call this API. See also: [AWS API Documentation](https://docs.aws.amazon.com/goto/WebAPI/cognito-identity-2014-06-30/GetId)